main/src/ca/uwaterloo/flix/language/ast/Ast.scala

```diff
@@ -446,4 +446,9 @@ object Ast {
     */
   case class ClassContext(superClasses: List[Symbol.ClassSym], instances: List[Ast.Instance])
 
+  /**
+    * Marks a value as metadata: the compiler must not rely on this information.
+    */
+  case class Meta[T](value: T)
```

Comment:
I don't like this _at all_ :)

Reply:
I understand why you want to do it, but what we really need is the Flix type system and row polymorphism etc.